Possible bug in aedit.c

First of all I tried to post this to the Development section of the Forum about 10 days ago. After doing the submit I got the message about the post will be first reviewed by a moderator. Well that was about 10 days ago and the posting has never appeared!! I did do a follow up posting to the General forum and that also never appeared!

I am building on Ubuntu 22.04.5 LTS and using GCC version - 11.4.0

I get the following warning: 
aedit.c: In function ‘aedit_parse’:
aedit.c:558:7: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  558 |       if (OLC_ACTION(d)->command)
      |       ^~
In file included from aedit.c:16:
oasis.h:135:24: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  135 | #define OLC_ACTION(d)  (OLC(d)->action)   /**< Action structure     */
      |                        ^
aedit.c:560:9: note: in expansion of macro ‘OLC_ACTION’
  560 |         OLC_ACTION(d)->command = strdup(arg);
      |         ^~~~~~~~~~

Looking at the code I see the following:

   case AEDIT_ACTION_NAME:
        ..
        ..
        ..
        ..
  
  Line 558        if (OLC_ACTION(d)->command)
                           free(OLC_ACTION(d)->command);
  Line 560 -->      OLC_ACTION(d)->command = strdup(arg);

                  break;
Further down the code I see this code:

case AEDIT_SORT_AS:
      if (!*arg || strchr(arg,' ')) {
        aedit_disp_menu(d);
        return;
      }
      if (OLC_ACTION(d)->sort_as) {
        free(OLC_ACTION(d)->sort_as);
        OLC_ACTION(d)->sort_as = strdup(arg);
      }
      break;

Note that the move of  strdup is inside the braces for the if.

For the if at line 558 there are no braces. I think that there should be and the move from strdup should be inside them.

Graham

--- a/src/aedit.c
+++ b/src/aedit.c
@@ -557,7 +557,7 @@ void aedit_parse(struct descriptor_data * d, char *arg) {
       }
       if (OLC_ACTION(d)->command)
         free(OLC_ACTION(d)->command);
-        OLC_ACTION(d)->command = strdup(arg);
+      OLC_ACTION(d)->command = strdup(arg);
 
       break;
 
@@ -566,10 +566,10 @@ void aedit_parse(struct descriptor_data * d, char *arg) {
         aedit_disp_menu(d);
         return;
       }
-      if (OLC_ACTION(d)->sort_as) {
+      if (OLC_ACTION(d)->sort_as) 
         free(OLC_ACTION(d)->sort_as);
-        OLC_ACTION(d)->sort_as = strdup(arg);
-      }
+      OLC_ACTION(d)->sort_as = strdup(arg);
+      
       break;
 
     case AEDIT_MIN_CHAR_POS:

